File: ndimclustering/calc_scale_mat.R
#' Rescale the columns of the matrix "mat"
#'
#' @description
#'   Rescale the columns of the matrix `mat` so that the values lie
#'   between -1 and 1 with the with the same distribution.
#'
#' @param mat The matrix whose columns are to be rescaled
#' @param narm Bool to indicate whether NaNs should be ignored in calculations.
#'   default \: TRUE
#'
#' @details Rescale of each column is found using \link{calc_col_scale} then
#'   column binding the results using \link[SparkR]{cbind} into "out_mat".
#'
#' @return out_mat
#'
#' @export
calc_scale_mat <- function(mat, narm = TRUE) {
  # Compute the Sample Mean with na_rm
  xbar <- apply(mat, 2, mean, na.rm = narm)
  # Compute the Sample SE with na_rm
  se <- apply(mat, 2, sd, na.rm = narm)
  out_list <- lapply(colnames(mat), calc_col_scale,
  data = mat, mu = xbar, se = se)
  out_mat <- Reduce(cbind, out_list)
  return(out_mat)
}

File: ndimclusteringpy/Python/calc_scale_mat.py
import pandas as pd
""" Rescale the columns of the matrix "mat"

   Rescale the columns of the matrix `mat` so that the values lie
   between -1 and 1 with the with the same distribution.

 :param mat: The matrix whose columns are to be rescaled
 :param narm: Bool to indicate whether NaNs should be ignored in calculations.
   default "TRUE"

  Rescale of each column is found using [calc_col_scale] then
   column binding the results using cbind into "out_mat".

 :return: out_mat"""
def calc_scale_mat (mat, narm = TRUE) :
  # Compute the Sample Mean with na_rm
  xbar = mat.mean(axis=1, skipna = narm)
  # Compute the Sample SE with na_rm
  se = mat.std(axis=1, skipna = narm)
  out_list = [calc_col_scale(c, data = mat, mu = xbar, se = se) for c in mat.columns]
  out_mat = pd.concat(out_list, axis = 1)
  return(out_mat)